src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -395,6 +398,40 @@ public function phpinfoAction()
     }
 
     /**
+     * Displays the source of a file.
+     *
+     * @return Response A Response instance
+     *
+     * @throws NotFoundHttpException
+     */
+    public function openAction(Request $request)
+    {
+        if (null === $this->kernelRootDir) {
+            throw new NotFoundHttpException('KernelRootDir should be set.');
+        }
+
+        if (null === $this->profiler) {
+            throw new NotFoundHttpException('The profiler must be enabled.');
```

Comment:
That's misleading. The profiler here should be available (a profile can to enabled or disabled as well, but that's something else).